capirca/lib/arista_tp.py

```diff
@@ -0,0 +1,1036 @@
+
+            term_block.append([MATCH_INDENT, "!", False])  # end of actions
+        term_block.append([TERM_INDENT, "!", False])  # end of match entry
+
+        for tindent, tstr, tverb in term_block:
+            config.Append(tindent, tstr, verbatim=tverb)
+
+        return str(config)
+
+    def _processPorts(self, term):
+        port_str = ""
+        sport = ""
+        dport = ""
+
+        # source port generation
+        if term.source_port:
+            sport += " source port %s" % self._Group(term.source_port)
+            port_str += sport
+
+        # destination port
+        if term.destination_port:
+            dport += (" destination port %s"
+                      % self._Group(term.destination_port))
+            port_str += dport
+
+        return port_str
+
+    def _processICMP(self, term):
+        icmp_types = [""]
+        icmp_code_str = ""
+        icmp_type_str = " type "
+
+        if term.icmp_type:
+            icmp_types = self.NormalizeIcmpTypes(
+                term.icmp_type, term.protocol, self.term_type
+            )
+        if icmp_types != [""]:
+            for t in icmp_types:
+                icmp_type_str += "%s," % t
+
+            if icmp_type_str.endswith(","):
+                icmp_type_str = icmp_type_str[:-1]  # chomp trailing ','
+                if not term.icmp_code:
+                    icmp_type_str += " code all"
+
+        if self.term.icmp_code and len(icmp_types) <= 1:
+            icmp_codes = self._Group(self.term.icmp_code)
+            icmp_codes = re.sub(r" ", ",", icmp_codes)
+            icmp_code_str += " code %s" % icmp_codes
+
+        return icmp_type_str, icmp_code_str
+
+    def _processProtocol(self, term_type, term, flags):
+        _ANET_PROTO_MAP = {
+            "inet": {
+                "ahp": "",
+                "bgp": "",
+                "icmp": "",
+                "igmp": "",
+                "ospf": "",
+                "pim": "",
+                "rsvp": "",
+                "tcp": "",
+                "udp": "",
+                "vrrp": "",
+            },
+            "inet6": {
+                # <0-255>    protocol  values(s) or range(s) of protocol  values
+                "bgp": "",        # BGP
+                "icmpv6": "",     # ICMPv6 (58)
+                "ospf": "",       # OSPF routing protocol (89)
+                "pim": "",        # Protocol Independent Multicast (PIM) (103)
+                "rsvp": "",       # Resource Reservation Protocol (RSVP) (46)
+                "tcp": "",        # TCP
+                "udp": "",        # UDP
+                "vrrp": "",       # VRRP (112)
+            }
+        }
+
+        protocol_str = ""
+        prots = []
+        # if there are dirty prots we'll need to convert the protocol list to
+        # all numbers and generate the list of protocols to match on. EOS
+        # doesn't support commingling named protocols w/numeric protocol-ids
```

Comment:
What does the device return if you do a "show config" type of command? 
When I input a:
1. protocol name
2. protocol numeric-id for a protocol that has a well known protocol name
3. protocol numeric-id for a protocol that doesn't have a name.

In order to meet this requirement [0], I'd recommend using what the device expects.

[0] - https://github.com/google/capirca/blob/master/doc/generator_patterns.md#syntax-of-the-config-from-the-generator-and-on-device-should-match-when-cryptographically-verified